### PR TITLE
Update GCP Development Machine Setup Documentation

### PR DESCRIPTION
**Pull request by pfandzelter**
Signed-off-by: Tobias Pfandzelter <pfandzelter@campus.tu-berlin.de>

# Reason for This PR

The setup documentation for development on Google Cloud was complete but had some subtle bugs and missing information that made it hard for newbies (like myself) to get started. Namely, information about availability of nested virtualization was missing, and some whitespace characters in code/shell listings made it difficult to copy/paste commands.

## Description of Changes

I have added the missing information and fixed formatting and whitespace to make everything a bit easier.
Alongside changes to the documentation I set up my own GCP development instance so I could verify everything.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
~~- [ ] Any newly added `unsafe` code is properly documented.~~
~~- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.~~
~~- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: docs/dev-machine-setup.md [hunk 1]
```diff
@@ -138,41 +138,49 @@ set up a Ubuntu-based VM on GCE with nested KVM enablement can be found in GCE
...
+    ```console
     $ FC_VDISK=disk-ub18
     $ FC_IMAGE=ub18-nested-kvm
-    $ gcloud compute disks create ${FC_VDISK}\
+    $ gcloud compute disks create ${FC_VDISK} \
     --image-project ubuntu-os-cloud --image-family ubuntu-1804-lts
-    $ gcloud compute images create ${FC_IMAGE} --source-disk ${FC_VDISK}\
-    --licenses "https://www.googleapis.com/compute/v1/projects/vm-options/global/licenses/enable-vmx"\
+    $ gcloud compute images create ${FC_IMAGE} --source-disk ${FC_VDISK} \
+    --licenses "https://www.googleapis.com/compute/v1/projects/vm-options/global/licenses/enable-vmx" \
```
Comment (dianpopa) [inline docs/dev-machine-setup.md:147]: Adding a space here makes the [markdown style](https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/style/test_markdown.py) test fail. Seems like a bug or limitation of the [mdl](https://github.com/markdownlint/markdownlint) linter. I suggest to remove the space for the moment. Otherwise, nice detailing of the GCP setup! Will approve once the tests are successful.

Comment (pfandzelter) [reply docs/dev-machine-setup.md]: That's an interesting corner case of mdl as described [here](https://github.com/markdownlint/markdownlint/issues/128#issuecomment-201106113_). The easiest solution will be to just move that particular line to the end of the command so it doesn't end in whitespace.

**Review by dianpopa — APPROVED** [review]

**Review by georgepisaltu — APPROVED** [review]
